src/uct/tcp/tcp_ep.c

```diff
@@ -350,21 +311,42 @@ static unsigned uct_tcp_ep_send(uct_tcp_ep_t *ep)
 
     iface->outstanding -= send_length;
     ep->tx.offset      += send_length;
-    if (ep->tx.offset == ep->tx.length) {
-        uct_tcp_ep_ctx_rewind(&ep->tx);
+    if (!uct_tcp_ep_ctx_buf_need_progress(&ep->tx)) {
+        uct_tcp_ep_ctx_reset(&ep->tx);
     }
 
     return send_length > 0;
 }
 
+static inline unsigned uct_tcp_ep_recv(uct_tcp_ep_t *ep, size_t *recv_length)
+{
+    ucs_status_t status;
+
+    ucs_assertv(recv_length > 0, "ep=%p", ep);
```

Comment:
`*recv_length`, but it is size_t, so maybe just check `ucs_assertv(*recv_length,...)`

Reply:
Good catch! Fixed